### DISCUSSION TITLE
Images Problem: Uncaught Mpdf\\MpdfException: Error detected

### DISCUSSION THREAD

**Discussion by craigbass76** [mpdf/mpdf#1968] [Q&A]
The whole error is this:
```
PHP Fatal error:  Uncaught Mpdf\\MpdfException: Error detected. PDF file generation aborted: fopen(/images/clockIn.png): Failed to open stream: No such file or directory in /var/www/html/documentation2/vendor/mpdf/mpdf/src/AssetFetcher.php:60\nStack trace:\n#0 /var/www/html/documentation2/createPDF.php(56): Mpdf\\Mpdf->Output()\n#1 {main}\n  thrown in /var/www/html/documentation2/vendor/mpdf/mpdf/src/AssetFetcher.php on line 60, referer: http://localhost/documentation2/docs/module/basics/testing
```

There is another image loading (`plus.png`) with no issues. It's sitting in the same directory, with the same permissions (755), and owned by the `www-data` user. The only difference I can see in the file is that it's smaller than `plus.png`, and the dimensions (ratio, really, `clockIn.png` is a landscape type pic, and `plus.png` is square) are different . They're both loading fine on the web page that I'm trying to "pdf-ize."

I also can't seem to load up any images in subdirectories. This html code fails in mPDF when I add the same image in a second line (the file is there - I can browse to `localhost://images/basics/plug.png` in chrome/firefox):
```
<p><img src="/images/plus.png" alt="" /></p>
<p><img src="/images/basics/plus.png" alt="" /></p>
```
There error on this is the same type - `...No such file...`

**Comment by finwe** [comment]
Are you generating the PDF through a browser? I'm guessing not?

**Reply by craigbass76** [reply]
I've got a link to `createPDF.php` from the page, and that script is doing the work. I thought that's what the docs were showing. There's a query running in `createPDF.php` to get the content, but the script is doign this at the end:
```
    require_once __DIR__ . '/vendor/autoload.php';
    $mpdf = new \Mpdf\Mpdf();
    $mpdf->debug = true;
    $mpdf->WriteHTML($docContentHTML, 2);
    $mpdf->Output('LBMLiftOff_UserManual-'.$moduleGetName.'_'.$docURL.'', 'D');
```

Wrong way to do it?

**Reply by finwe** [reply]
Are you running the script in the browser, on in the command line? What is the content of 
$_SERVER['HTTP_HOST'], $_SERVER['SERVER_NAME'], $_SERVER['SCRIPT_NAME'] supervariables? What is the content of $mpdf->basepath? It seems that mPDF interprets the basepath to be local file path and looks for the image in your / filesystem instead of relative to the http host/path.

**Reply by craigbass76** [reply]
This:
```
echo $_SERVER['HTTP_HOST'];
echo '<br />';
echo $_SERVER['SERVER_NAME'];
echo '<br />';
echo $_SERVER['SCRIPT_NAME'] ;
```
Gives me this (viewing source on rendered html page):
```
localhost<br />
localhost<br />
/documentation2/createPDF.php
```

**Reply by finwe** [reply]
What is the content of $mpdf->basepath?

**Reply by finwe** [reply]
Also, enable [logging](https://mpdf.github.io/installation-setup/logging.html) and see/post logger output
